nova_vm/src/ecmascript/abstract_operations/testing_and_comparison.rs

```diff
@@ -277,3 +277,137 @@ pub(crate) fn is_less_than<const LEFT_FIRST: bool>(
         Ok(Some(rnx < rny))
     }
 }
+
+/// [7.2.14 IsLooselyEqual ( x, y )](https://tc39.es/ecma262/#sec-islooselyequal)
+///
+/// The abstract operation IsLooselyEqual takes arguments x (an ECMAScript
+/// language value) and y (an ECMAScript language value) and returns either a
+/// normal completion containing a Boolean or a throw completion. It provides
+/// the semantics for the == operator.
+pub(crate) fn is_loosely_equal(
+    agent: &mut Agent,
+    x: impl Into<Value> + Copy,
+    y: impl Into<Value> + Copy,
+) -> JsResult<bool> {
+    let (x, y) = (x.into(), y.into());
```

Comment:
What do you mean by this? Like this?:
```ts
    fn inner(x: Value, y: Value) {
      ...
    }
    
    return inner(x.into(), y.into());
```

Reply:
Do [these changes](https://github.com/trynova/nova/pull/53/commits/773d8cb13216e56bf75b5ddbbe1607d69584d03a) do what you mean?

Reply:
The point sort of is that the entire function is generic for the parameters. It's not polymorphism but specialised monomorphism.

Reply:
Discussed on Discord, I prefer the original way (keep the generics for the whole function in hopes of better compiler code generation), @vimirage prefers the suggested way (use separate internal function so that the function doesn't get compiled separately for each possible set of generic parameters but still allows for the convenience of not having to call function with `func(a.into(), b.into())` every time).

So, do as you prefer @eliassjogreen :)

Reply:
I will do as you prefer, I find it easier to read and hopefully LLVM is kind to us when it comes to codegen.